src/inmanta/moduletool.py

```diff
@@ -251,17 +253,25 @@ def diff(cls, *, low: Version, high: Version) -> Optional["ChangeType"]:
             return cls.MINOR
         if high.micro > low.micro:
             return cls.PATCH
+        if len(high.base_version.split('.')) >= 4:
+            high_revision = high.base_version.split('.')[-1]
+            # We are switching from 3 digits to 4
+            if len(low.base_version.split('.')) < 4 or \
+                (len(low.base_version.split('.')) >= 4 and high_revision > low.base_version.split('.')[-1]):
```

Comment:
This part seems redundant to me
```suggestion
            if len(low.base_version.split('.')) < 4 or high_revision > low.base_version.split('.')[-1]:
```